btree/btree.c

```diff
@@ -2039,36 +2091,149 @@ static void generic_move(struct nd *src, struct nd *tgt,
 	node_fix(tgt, tx);
 }
 
+/**
+ * calc_shift is used to calculate the shift for the given number of bytes.
+ * Shift is the exponent of nearest power-of-2 value greater than or equal to
+ * number of bytes. 
+ * 
+ * @param value It represents the number of bytes
+ * @return int returns the shift value.
+ */
+
+int calc_shift(int value)
+{
+	unsigned int sample 	= (unsigned int) value;
+	unsigned int pow 	= 0;
+
+	while (sample > 0)
+	{
+		sample >>=1;
+		pow += 1;
+	}
+
+	return pow - 1;
+}
+
+/**
+ * btree_create_tick function is the main function used to create btree.
+ * It traverses through multiple states to perform its operation.
+ * 
+ * @param smop Represents the state machine operation
+ * @return int64_t It returns the next state to be executed.
+ */
+
+int64_t btree_create_tick(struct m0_sm_op *smop)
+{
+	struct m0_btree_op 	*bop = M0_AMB(bop,smop,bo_op);
+	struct m0_btree_oimpl 	*oi = bop->bo_i;
+	struct td		*tree;
+	struct m0_btree_idata 	*data;
+	struct segaddr 		 curr_addr;
+	int 			 k_size = 8;
```

Comment:
why the values are hardcoded ?

Reply:
At this point, we dont have the mechanism to get key size and value size.

Reply:
you have enum values MAX_KEY_SIZE, MAX_VAL_SIZE for now

Reply:
They are not part of this PR. Hence, I didnt use them.
Changes have been made as per shashank's suggestion.